types/dom-mediacapture-record/index.d.ts

```diff
@@ -48,6 +49,13 @@ declare class MediaRecorder extends EventTarget {
 
     constructor(stream: MediaStream, options?: MediaRecorderOptions);
 
+    addEventListener(type: 'dataavailable', listener: (this: MediaRecorder, event: BlobEvent) => void, useCapture?: boolean): void;
```

Comment:
I don't think this is needed. MediaRecorder already extends EventTarget which has the correct type for `addEventListener`.

Reply:
Right, but it lacks the typings for `'dataavailable'` and `'error'` events

Reply:
Ah I see, good improvement! maybe you could mimic the way TypeScript does it: https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L2381-L2400. They use an EventMap to enforce both type and event.

Reply:
That's what I did locally in my own projects, I'll gladly do so 😄